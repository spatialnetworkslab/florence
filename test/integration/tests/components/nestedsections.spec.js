/// <reference types="Cypress" />

context('Assertions', () => {
  beforeEach(() => {
    cy.visit('/nestedsections')
  })

  describe('Nested Section Tests', () => {
    it('[render] nested sections rendered', () => {
      cy.get('.section > .section').should('have.length', 3)
    })

    it('[render] nested sections positioned correctly', () => {
      cy.get('.section > .section > .content-background').each(($el, idx) => {
        const midpoint = +$el.attr('x') + (+$el.attr('width') / 2)

        cy.get('.x-axis > .label-layer > .label').then($arr => {
          const labelXPos = +Cypress.dom.wrap($arr[idx]).attr('x')
          expect(labelXPos).to.be.closeTo(midpoint, 0.01)
        })
      })
    })

    // it('[render] nested sections snapshot correctly', () => {
    //   cy.get('svg').toMatchSnapshot()
    // })
  })
})
