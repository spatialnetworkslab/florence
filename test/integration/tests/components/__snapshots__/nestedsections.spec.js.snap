exports[`Assertions > Nested Section Tests > [render] nested sections snapshot correctly #0`] = `
<svg
  id="gr0"
  width="600"
  height="430"
  viewBox="0 0 600 430"
  preserveAspectRatio="xMidYMid meet"
>
  <defs>
    <mask id="gr0-mask-padding-bg">
      <rect x="0" y="0" width="600" height="430" fill="white"></rect>
      <rect x="0" y="0" width="600" height="430" fill="black"></rect>
    </mask>
  </defs>
  <defs>
    <clipPath id="clip-sc0">
      <rect x="0" y="0" width="600" height="430"></rect>
    </clipPath>
    <mask id="mask-sc0-padding-bg">
      <rect x="0" y="0" width="600" height="430" fill="white"></rect>
      <rect x="30" y="30" width="540" height="370" fill="black"></rect>
    </mask>
  </defs>
  <g class="section">
    <rect
      class="content-background"
      x="30"
      y="30"
      width="540"
      height="370"
      fill="#6e75a8"
    ></rect>
    <defs>
      <clipPath id="clip-sc1">
        <rect
          x="47.419354838709694"
          y="30"
          width="156.7741935483871"
          height="370"
        ></rect>
      </clipPath>
      <mask id="mask-sc1-padding-bg">
        <rect
          x="47.419354838709694"
          y="30"
          width="156.7741935483871"
          height="370"
          fill="white"
        ></rect>
        <rect
          x="47.419354838709694"
          y="30"
          width="156.7741935483871"
          height="370"
          fill="black"
        ></rect>
      </mask>
    </defs>
    <g class="section">
      <rect
        class="content-background"
        x="47.419354838709694"
        y="30"
        width="156.7741935483871"
        height="370"
        fill="#dbbbf5"
      ></rect>
    </g>
    <defs>
      <clipPath id="clip-sc2">
        <rect
          x="221.61290322580646"
          y="30"
          width="156.77419354838713"
          height="370"
        ></rect>
      </clipPath>
      <mask id="mask-sc2-padding-bg">
        <rect
          x="221.61290322580646"
          y="30"
          width="156.77419354838713"
          height="370"
          fill="white"
        ></rect>
        <rect
          x="221.61290322580646"
          y="30"
          width="156.77419354838713"
          height="370"
          fill="black"
        ></rect>
      </mask>
    </defs>
    <g class="section">
      <rect
        class="content-background"
        x="221.61290322580646"
        y="30"
        width="156.77419354838713"
        height="370"
        fill="#dbbbf5"
      ></rect>
    </g>
    <defs>
      <clipPath id="clip-sc3">
        <rect
          x="395.80645161290323"
          y="30"
          width="156.77419354838702"
          height="370"
        ></rect>
      </clipPath>
      <mask id="mask-sc3-padding-bg">
        <rect
          x="395.80645161290323"
          y="30"
          width="156.77419354838702"
          height="370"
          fill="white"
        ></rect>
        <rect
          x="395.80645161290323"
          y="30"
          width="156.77419354838702"
          height="370"
          fill="black"
        ></rect>
      </mask>
    </defs>
    <g class="section">
      <rect
        class="content-background"
        x="395.80645161290323"
        y="30"
        width="156.77419354838702"
        height="370"
        fill="#dbbbf5"
      ></rect>
    </g>
    <g class="x-axis">
      <g clip-path="none" class="line-layer">
        <path
          class="line"
          d="M125.81,400L125.81,405"
          fill="none"
          stroke-width="0.5"
          stroke="black"
          style="opacity: 1"
        ></path>
        <path
          class="line"
          d="M300,400L300,405"
          fill="none"
          stroke-width="0.5"
          stroke="black"
          style="opacity: 1"
        ></path>
        <path
          class="line"
          d="M474.19,400L474.19,405"
          fill="none"
          stroke-width="0.5"
          stroke="black"
          style="opacity: 1"
        ></path>
      </g>
      <g clip-path="none" class="label-layer">
        <text
          class="label"
          x="125.80645161290325"
          y="407.00000000000006"
          fill="black"
          stroke="none"
          stroke-width="0"
          opacity="1"
          transform="
          rotate(0, 
          125.80645161290325, 
          407.00000000000006)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="middle"
          dominant-baseline="hanging"
        >
          iris
        </text>
        <text
          class="label"
          x="300"
          y="407.00000000000006"
          fill="black"
          stroke="none"
          stroke-width="0"
          opacity="1"
          transform="
          rotate(0, 
          300, 
          407.00000000000006)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="middle"
          dominant-baseline="hanging"
        >
          versicolor
        </text>
        <text
          class="label"
          x="474.1935483870967"
          y="407.00000000000006"
          fill="black"
          stroke="none"
          stroke-width="0"
          opacity="1"
          transform="
          rotate(0, 
          474.1935483870967, 
          407.00000000000006)
        "
          font-family="Helvetica"
          font-size="10px"
          font-weight="normal"
          text-anchor="middle"
          dominant-baseline="hanging"
        >
          virginica
        </text>
      </g>
    </g>
  </g>
</svg>
`;
